### DISCUSSION TITLE
Input widget and auto focus disabling

### DISCUSSION THREAD

**Discussion by blackdwarf** [Textualize/textual#4143] [Q&A] [answered]
Hi,

I've started playing with Textual and am making a small app with a UI that is split into three boxes, two on the left side and one on the right side. You can see this in the attached image.

![image](https://github.com/Textualize/textual/assets/3115004/9f901890-b5d3-40cd-983c-0112952bb019)

When the app starts, however, the search input box automatically gets focus, and so it is hard to test out other bindings. 

Is there a way to stop this from happening? I've added the button below so I can tab out and continue testing, but for the life of me I don't know how to just have it not take focus.

**Comment by TomJGooding** [comment, marked as answer]
You can disable auto-focus on the App or Screen with the `AUTO_FOCUS` attribute.

- https://textual.textualize.io/api/app/#textual.app.App.AUTO_FOCUS
- https://textual.textualize.io/api/screen/#textual.screen.Screen.AUTO_FOCUS

```python
from textual.app import App, ComposeResult
from textual.widgets import Input


class NoAutoFocusApp(App):
    AUTO_FOCUS = ""

    def compose(self) -> ComposeResult:
        yield Input()


if __name__ == "__main__":
    app = NoAutoFocusApp()
    app.run()
```

**Comment by blackdwarf** [comment]
Well, I knew that there had to be _something_ in the framework for this. Thank you @TomJGooding this works!
